lib/ecto/pools/ownership.ex

```diff
@@ -0,0 +1,202 @@
+defmodule Ecto.Pools.Ownership do
+  defmodule Strategy do
+    use Behaviour
+    alias Ecto.Pool
+
+    defcallback ownership_checkout(module, pid) :: Pool.mode
+    defcallback ownership_checkin(module, pid) :: Pool.mode
+  end
+
+  defmodule Server do
+    use GenServer
+
+    @timeout 5_000
```

Comment:
When we call `__pool__`, the third argument is a timeout. Let's use that instead of the hardcoded value.